### PR TITLE
Fixing exception when config is missing due to relock of mutex

### PR DESCRIPTION
**Pull request by jasonsandlin**

### CODE REVIEW

File: Source/Shared/local_config.cpp [hunk 1]
```diff
@@ -20,21 +20,28 @@ NAMESPACE_MICROSOFT_XBOX_SERVICES_CPP_BEGIN
...
-    if (xsapiSingleton->s_localConfigSingleton == nullptr)
-    {
-        xsapiSingleton->s_localConfigSingleton = std::make_shared<local_config>();
-
+	bool needToReadConfig = false;
+	{
+		std::lock_guard<std::mutex> guard(xsapiSingleton->s_singletonLock);
+		if (xsapiSingleton->s_localConfigSingleton == nullptr)
+		{
+			needToReadConfig = true;
```
Comment (edjamesmsft) [inline Source/Shared/local_config.cpp]: >			 [](start = 0, length = 3)

tabs

File: Source/Shared/local_config.cpp [hunk 2]
```diff
@@ -20,21 +20,28 @@ NAMESPACE_MICROSOFT_XBOX_SERVICES_CPP_BEGIN
...
+		std::lock_guard<std::mutex> guard(xsapiSingleton->s_singletonLock);
+		if (xsapiSingleton->s_localConfigSingleton == nullptr)
+		{
+			needToReadConfig = true; 
+			xsapiSingleton->s_localConfigSingleton = std::make_shared<local_config>();
+		}
+	}
+
+	if (needToReadConfig)
+	{
```
Comment (edjamesmsft) [inline Source/Shared/local_config.cpp]: if no action for TV_API, move this if statement inside the #if

Comment (jasonsandlin) [inline Source/Shared/local_config.cpp]: the issue is then needToReadConfig gets warnings as unused in TV_API builds

Comment (edjamesmsft) [inline Source/Shared/local_config.cpp]: ah okay, fair enough :)

---
In reply to: [107504772](https://github.com/Microsoft/xbox-live-api/pull/108#discussion_r107504772) [](ancestors = 107504772)

**Pull request merged**